source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -243,6 +243,8 @@ the function implementation the driver MUST:
 - Drivers MUST record the durations and counts of the heartbeats, the durations of the
   commands, as well as keep track of the number of open connections, and report this information in
   the function response as JSON.
+- Drivers MUST assert no ServerHeartbeat events contain the ``awaited=True`` flag to
```

Comment:
Do you think it's worth extending the test plan for handshakes (https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst#test-plan) to check that the the initial handshake uses OP_QUERY? As of this ticket, there is a possibility that polling could be used with a server versions >= 4.4 . And AFAIK this is novel and so potentially important to ensure that correct handshake logic is being implemented.

It's probably not very important, as the wire message between streaming / polling are not that different. But I do notice we don't check this anywhere and so it could be useful to do so as a part of this change.

Reply:
I think this idea is worthwhile but as a new drivers ticket since this PR doesn't change the OP_QUERY vs OP_MSG behavior.